pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -95,6 +96,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.RetentionDeleteWorkCount, "boltdb.shipper.compactor.retention-delete-worker-count", 150, "The total amount of worker to use to delete chunks.")
 	f.DurationVar(&cfg.DeleteRequestCancelPeriod, "boltdb.shipper.compactor.delete-request-cancel-period", 24*time.Hour, "Allow cancellation of delete request until duration after they are created. Data would be deleted only after delete requests have been older than this duration. Ideally this should be set to at least 24h.")
 	f.IntVar(&cfg.MaxCompactionParallelism, "boltdb.shipper.compactor.max-compaction-parallelism", 1, "Maximum number of tables to compact in parallel. While increasing this value, please make sure compactor has enough disk space allocated to be able to store and compact as many tables.")
+	f.IntVar(&cfg.UploadParallelism, "boltdb.shipper.compactor.upload-parallelism", uploadIndexSetsConcurrency, "Number of upload/remove operations to execute in parallel when finalizing a compaction. Parallel compactions will each issue this many uploads concurrently.")
```

Comment:
I think we should drop the constant and hardcode it here, which is easy to reference.

We can also drop `Parallel compactions will each issue this many uploads concurrently.` from the help text to keep it concise.